Add support for generating IdToken and UserInfo example payloads (#411)

thomasdarimont opened the pull request:
Fixes #411

See: https://github.com/keycloak/keycloak/blob/main/services/src/main/java/org/keycloak/services/resources/admin/ClientScopeEvaluateResource.java#L159

Signed-off-by: Thomas Darimont <thomas.darimont@googlemail.com>

ssilvert commented:
Also related to https://github.com/keycloak/keycloak-admin-ui/issues/1564

jonkoops commented on src/resources/clients.ts:
```diff
@@ -411,6 +411,27 @@ export class Clients extends Resource<{realm?: string}> {
 
+  public evaluateGenerateIdToken = this.makeRequest<
+  {id: string; scope: string; userId: string},
+  object
```
According to the source the return type for this one is `Map<String, Object>`, so we can use a more narrow type here in TypeScript as well.
```suggestion
  Record<string, unknown>
```

jonkoops commented on src/resources/clients.ts:
```diff
@@ -411,6 +411,27 @@ export class Clients extends Resource<{realm?: string}> {
+
+  public evaluateGenerateUserInfo = this.makeRequest<
+  {id: string; scope: string; userId: string},
+  object
```
This method seems to return a type of [`IdToken`](https://www.keycloak.org/docs-api/16.1/rest-api/index.html#_idtoken), let's make this reflect that.

thomasdarimont commented:
@jonkoops thanks for taking a look, how about returning Record<string, unknown> from both methods and add the more precise type information later?

I started to create the proper type definitions for JsonWebToken, IDToken, AccessToken and UserInfo - saved those in [this gist](https://gist.github.com/thomasdarimont/d8272edb303bda2d1ddfc93a26512a5a).
I was not sure whether this was necessary as there is already `AccessTokenRepresentation.ts`. Perhaps you can take it from there :)

jonkoops commented:
@thomasdarimont could you give this PR a rebase? Our CI was borked for a while but should be good now.

thomasdarimont commented:
@jonkoops thanks for the feedback. I just rebased and updated the PR. I noticed that we now already have a `evaluateGenerateIdToken` and `evaluateGenerateUserInfo` function.

The only effective change this PR brings is changing the return type from `object` to `Record<string, unknown>` which might require projects to change code.

Hence, I think this PR is obsolete now and can be closed.
